app/Http/Controllers/V2/Terrafund/TerrafundCreateGeometryController.php

```diff
@@ -0,0 +1,757 @@
+            }
+        } else {
+            return response()->json(['error' => 'No file uploaded'], 400);
+        }
+    }
+
+    private function findShpFile($directory)
+    {
+        $shpFile = null;
+        $files = scandir($directory);
+        foreach ($files as $file) {
+            if (pathinfo($file, PATHINFO_EXTENSION) === 'shp') {
+                $shpFile = "{$directory}/{$file}";
+
+                break;
+            }
+        }
+
+        return $shpFile;
+    }
+
+    public function checkSelfIntersection(Request $request)
+    {
+        $uuid = $request->query('uuid');
+        $geometry = PolygonGeometry::where('uuid', $uuid)->first();
+
+        if (! $geometry) {
+            return response()->json(['error' => 'Geometry not found'], 404);
+        }
+
+        $isSimple = DB::selectOne('SELECT ST_IsSimple(geom) AS is_simple FROM polygon_geometry WHERE uuid = :uuid', ['uuid' => $uuid])->is_simple;
+        $SELF_CRITERIA_ID = 4;
+        $message = $isSimple ? 'The geometry is valid' : 'The geometry has self-intersections';
+        $insertionSuccess = $this->insertCriteriaSite($uuid, $SELF_CRITERIA_ID, $isSimple);
+
+        return response()->json(['selfintersects' => $message, 'geometry_id' => $geometry->id, 'insertion_success' => $insertionSuccess, 'valid' => $isSimple ? true : false], 200);
+    }
+
+    public function calculateDistance($point1, $point2)
+    {
+        $lat1 = $point1[1];
+        $lon1 = $point1[0];
+        $lat2 = $point2[1];
+        $lon2 = $point2[0];
+
+        $theta = $lon1 - $lon2;
+        $dist = sin(deg2rad($lat1)) * sin(deg2rad($lat2)) + cos(deg2rad($lat1)) * cos(deg2rad($lat2)) * cos(deg2rad($theta));
+        $dist = acos($dist);
+        $dist = rad2deg($dist);
+        $miles = $dist * 60 * 1.1515;
+
+        return $miles * 1.609344; // Convert to kilometers
+    }
+
+    // public function detectSpikes($geometry)
+    // {
+    //   $spikes = [];
+    //   if ($geometry['type'] === 'Polygon' || $geometry['type'] === 'MultiPolygon') {
+    //     $coordinates = $geometry['type'] === 'Polygon' ? $geometry['coordinates'][0] : $geometry['coordinates'][0][0]; // First ring of the polygon or the first polygon in the MultiPolygon
+    //     $numVertices = count($coordinates);
+    //     $perimeter = 0;
+    //     for ($i = 0; $i < $numVertices - 1; $i++) {
+    //       $distance = $this->calculateDistance($coordinates[$i], $coordinates[$i + 1]);
+    //       $perimeter += $distance;
+    //     }
+    //     $averageDistance = $perimeter / ($numVertices - 1);
+    //     $threshold = $averageDistance * 1.25; // Adjust this threshold as needed
+    //     for ($i = 0; $i < $numVertices - 1; $i++) {
+    //       $distance = $this->calculateDistance($coordinates[$i], $coordinates[$i + 1]);
+    //       if (abs($distance - $averageDistance) > $threshold) {
+    //         $spikes[] = $coordinates[$i];
+    //       }
+    //     }
+    //   }
+    //   return $spikes;
+    // }
```

Comment:
I'd like no commented out code to be committed to `staging`. Will this method be resurrected? I'm guessing not since there's another implementation below.

Reply:
removed